h2o-persist-gcs/src/main/java/water/persist/GcsBlob.java

```diff
@@ -0,0 +1,96 @@
+package water.persist;
+
+import com.google.cloud.storage.Blob;
+import com.google.cloud.storage.BlobId;
+import com.google.cloud.storage.BlobInfo;
+import water.Key;
+import water.api.FSIOException;
+import water.fvec.Vec;
+
+import java.net.URI;
+import java.util.Arrays;
+
+class GcsBlob {
+  static final String KEY_PREFIX = "gs://";
+  private static final int KEY_PREFIX_LENGTH = KEY_PREFIX.length();
+  private final String canonical;
+  private final BlobId blobId;
+  private final Key key;
+
+  private GcsBlob(String bucket, String key) {
+    canonical = toCanonical(bucket, key);
+    blobId = BlobId.of(bucket, key);
+    this.key = Key.make(canonical);
+  }
+
+  static GcsBlob of(String bucket, String key) {
```

Comment:
A missing constructor replaced by a factory method could be misleading for some, but this one is intended for internal use only. Not a blocker.